core/trace/dt_span_serializer_mapper.go

```diff
@@ -159,21 +160,31 @@ func getProtoEvents(events []sdktrace.Event) ([]*protoTrace.Span_Event, error) {
 	return protoEvents, nil
 }
 
-func getProtoLinks(links []sdktrace.Link) ([]*protoTrace.Span_Link, error) {
+func getProtoLinks(links []sdktrace.Link, tenantId, clusterId int32) ([]*protoTrace.Span_Link, error) {
 	protoLinks := make([]*protoTrace.Span_Link, 0, len(links))
 	for _, link := range links {
-		traceId := link.SpanContext.TraceID()
-		spanId := link.SpanContext.SpanID()
+		spanContext := link.SpanContext
+		traceId := spanContext.TraceID()
+		spanId := spanContext.SpanID()
 		protoAttributes, err := getProtoAttributes(link.Attributes)
 		if err != nil {
 			return nil, err
 		}
+
 		protoLink := &protoTrace.Span_Link{
 			TraceId:                traceId[:],
 			SpanId:                 spanId[:],
 			Attributes:             protoAttributes,
 			DroppedAttributesCount: uint32(link.DroppedAttributeCount),
 		}
+
+		if spanContext.IsRemote() {
+			if fw4Tag, err := fw4.GetMatchingFw4FromTracestate(spanContext.TraceState(), tenantId, clusterId); err != nil {
```

Comment:
since go works with composition, i wonder if returning custom tracestate containing the parsed fw4 tag in the propagator would also work. prerequisite would be that the same tracestate instance is passed to all child spans. 
this would avoid re-parsing the tracestate here for every link

Reply:
Interesting idea! I'll look into it.

Reply:
I couldn't figure out a way to make this work. The `TraceState` that is part of the `SpanContext` is not composable, and neither is the `SpanContext` itself (the otel SDK works directly with those struct types). The only way to do composition here, that I can think of, would be to store the Fw4Tag on the Go Context, which we already do. However we do not have access to the Context associated with a specific SpanContext anymore at the time of serialization.